tests/functional/books.spec.ts

```diff
@@ -57,4 +58,28 @@ test.describe(`Books tests`, () => {
       await profilePage.getBookRow(0).verifyBook(bookForTest);
     });
   });
+
+  test('Search book test', async ({bookStorePage}) => {
+    const searchText = 'JavaScript'
+    const allBookTitles = await bookStorePage.bookTitles.allTextContents();
+    const expectedFilteredBookTitles = allBookTitles.filter(item => item.includes(searchText));
+    await bookStorePage.textSearchInput.fill(searchText);
+    const actualFilteredBookTitles = await bookStorePage.bookTitles.allTextContents();
+    expect(JSON.stringify(actualFilteredBookTitles)).toBe(JSON.stringify(expectedFilteredBookTitles));
+  })
+  test('Add book to profile', async ({bookStorePage, profilePage}) => {
+    const allBookTitles = await bookStorePage.bookTitles.allTextContents();
+    const bookForTest = allBookTitles[bookStorePage.getRandomInt(allBookTitles.length)];
+    await bookStorePage.getBookLink(bookForTest).click();
+    await bookStorePage.addBookToCollection.click();
+    await profilePage.open();
+    const profileBookTitles = await bookStorePage.bookTitles.allInnerTexts();
+    await expect(profileBookTitles).toContain(bookForTest);
+    const deleteSelector = '//..//..//..//span[@title="Delete"]'
```

Comment:
1. Try to avoid such long selectors like this , because a) not readable b) tomorrow developer may add some additional <div>  to page and this complex locator will easily break
2. Locators should be placed in Page object

Reply:
3. //span[contains(@id, "see-book-Git Pocket Guide")]//ancestor::div[@role="rowgroup"]//span[@title="Delete"]
this one better